src/three-components/CachingGLTFLoader.ts

```diff
@@ -94,28 +141,42 @@ export class CachingGLTFLoader {
     if (model != null) {
       model.userData.animations = gltf.animations;  // save animations
 
-      // This is a patch to three's handling of PMREM environments. This patch
-      // has to be applied after cloning because three does not seem to clone
-      // the onBeforeCompile method.
-      const updateShader = (shader: Shader) => {
-        shader.fragmentShader =
-            shader.fragmentShader
-                .replace('#include <cube_uv_reflection_fragment>', cubeUVChunk)
-                .replace(
-                    '#include <envmap_physical_pars_fragment>', envmapChunk);
-      };
-
-      model.traverse((node: any) => {
-        if (Array.isArray(node.material)) {
-          (node.material as Array<Material>).forEach(material => {
-            material.onBeforeCompile = updateShader;
-          });
-        } else if (node.material != null) {
-          node.material.onBeforeCompile = updateShader;
-        }
+      this[$evictionPolicy].retain(url);
+
+      Object.defineProperty(model, $releaseFromCache, {
+        value: (() => {
```

Comment:
This doesn't seem like our usual style; what's the purpose of Object.defineProperty here?

Reply:
It's the only way I could get `tsc` to stop complaining about me decorating this object.

It might be possible to change the type of the identifier and get a cleaner result. I'll try that out.